Install/install.sh

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash -e
+
+GIT_REPO=https://github.com/TomWhitwell/SlowMovie
+GIT_BRANCH=master
+PYTHON_VERSION=3
+SKIP_DEPS=false
+
+function install_linux_packages(){
+  sudo apt-get update
+
+  sudo apt-get install -y ffmpeg git python3-pip
+}
+
+function install_python_packages(){
+  # set the python3 as the default version
+  sudo update-alternatives --install /usr/bin/python python /usr/bin/python3 1
+  sudo update-alternatives --set python /usr/bin/python${PYTHON_VERSION}
+
+
+  sudo pip3 install setuptools -U
+  sudo pip3 install -r $LOCAL_DIR/Install/requirements.txt
+}
+
+function build_python_libraries(){
+  WAVESHARE_DIR=/home/pi/e-Paper
+
+  if [ -d "${WAVESHARE_DIR}" ]; then
+    echo -e "Updating Waveshare Drivers"
+    cd $WAVESHARE_DIR
+    git fetch
+    git pull
+  else
+    echo -e "Installing Waveshare Drivers"
+    git clone https://github.com/waveshare/e-Paper
+  fi
+
+  echo -e "$(tput setaf 1)Be patient - this takes time$(tput sgr 0)"
+  cd $WAVESHARE_DIR/RaspberryPi_JetsonNano/python/
+  sudo python3 setup.py install
+
+  # return to home directory
+  cd /home/pi
+}
+
+function setup_hardware(){
+  echo "Setting up SPI"
+  if ls /dev/spi* &> /dev/null; then
+      echo -e "SPI already enabled"
+  else
+      if command -v raspi-config > /dev/null && sudo raspi-config nonint get_spi | grep -q "1"; then
+          sudo raspi-config nonint do_spi 0
+          echo -e "SPI is now enabled"
+      else
+          echo -e "$(tput setaf 1)There was an error enabling SPI, enable manually with sudo raspi-config$(tput sgr 0)"
+      fi
+  fi
+}
+
+function install_slowmovie(){
+  if [ "$SKIP_DEPS" = false ]; then
+    # install from apt
+    install_linux_packages
+
+    # configure the hardware
+    setup_hardware
+  else
+    echo -e "Skipping dependency installs, updating SlowMovie code only"
+  fi
+
+  if [ -d "${LOCAL_DIR}" ]; then
+    echo -e "Existing Install - Updating Repo"
+  else
+    echo -e "No Install Found - Cloning Repo"
+    git clone ${GIT_REPO} ${LOCAL_DIR}
+  fi
+
+  # update the repo
+  cd $LOCAL_DIR
+  git fetch
+  git checkout $GIT_BRANCH
+  git pull
```

Comment:
```suggestion
  # update the repo
  cd $LOCAL_DIR
  git pull
else
  echo -e "No Install Found - Cloning Repo"
  git clone ${GIT_REPO} ${LOCAL_DIR}
  fi
```
Only `git pull` is needed to update, and there's no point pulling immediately after cloning.

Reply:
I want to keep the `fetch` and the `checkout` as it is possible to switch branches. You could envision a scenario where someone might be running off a branch and then want to switch to a different one. I did move some things around to checkout the branch specifically with `git clone -b` and then move the other commands to within the IF statement.